Tasks/NuGetInstaller/VsoNuGetHelper.ps1

```diff
@@ -0,0 +1,163 @@
+    [void]$passwordElement.SetAttribute("key", "Password")
+    $encryptedPassword = EncryptNuGetPassword $password
+    [void]$passwordElement.SetAttribute("value", $encryptedPassword)
+    [void]$newCredential.AppendChild($passwordElement)
+
+    [void]$credentialsSection.AppendChild($newCredential)
+}
+
+function ShouldGetCredentialsForFeed
+{
+    [CmdletBinding()]
+    param(
+        [Uri]$feedUri
+    )
+
+    if(-not ($feedUri.Scheme.Equals("http","InvariantCultureIgnoreCase") -or  $feedUri.Scheme.Equals("https","InvariantCultureIgnoreCase")))
+    {
+        return $false
+    }
+
+    $WebRequestObject = [System.Net.HttpWebRequest] [System.Net.WebRequest]::Create($feedUri);
+    try
+    {
+        $ResponseObject = [System.Net.HttpWebResponse] $WebRequestObject.GetResponse();
+    }
+    catch [Net.WebException] 
+    {
+        $ResponseObject = [System.Net.HttpWebResponse] $_.Exception.Response
+    }
+    
+    if($ResponseObject.StatusCode -eq 401)
+    {
+        $tfsHeaderPresent = $false
+        $vssHeaderPresent = $false
+        foreach($key in $ResponseObject.Headers.AllKeys)
+        {
+            if($tfsHeaderPresent -or $key.Contains("X-TFS"))
+            {
+                $tfsHeaderPresent = $true    
+            }
+
+            if($vssHeaderPresent -or $key.Contains("X-VSS"))
+            {
+                $vssHeaderPresent = $true
+            }
+
+            if($tfsHeaderPresent -and $vssHeaderPresent)
+            {
+                break
+            }
+        }
+
+        [void]$ResponseObject.Close();
+        return $tfsHeaderPresent -and $vssHeaderPresent
+    }
+    
+    return $false
+}
+
+function SetCredentialsNuGetConfigAndSaveTemp
+{
+    [CmdletBinding()]
+    param(
+        [xml] $nugetConfig,
+        [string] $accessToken,
+        [uri]$targetFeed
+    )
+
+    $credentialsSection = $nugetConfig.SelectSingleNode("configuration/packageSourceCredentials")
+    if($credentialsSection -eq $null)
+    {
+        #add the packageSourceCredentials node
+        $credentialsSection = $nugetConfig.CreateElement("packageSourceCredentials")
+        [void]$nugetConfig.configuration.AppendChild($credentialsSection)
+    }
+    
+    $nugetSources = $nugetConfig.SelectNodes("configuration/packageSources/add")
+    foreach($nugetSource in $nugetSources)
+    {
+        if($targetFeed -and (-not $targetFeed.ToString().Equals($nugetSource.value,"InvariantCultureIgnoreCase")))
+        {
+            continue
+        }
+
+        if(-not (ShouldGetCredentialsForFeed $nugetSource.value))
+        {
+            continue
+        }
+
+        if($credentialsSection.SelectSingleNode([string]$nugetSource.key) -ne $null)
+        {
+            $credentials = $credentialsSection.SelectSingleNode([string]$nugetSource.key)
+            foreach($section in $credentials.SelectNodes("add"))
+            {
+                if(([System.Xml.XmlNode]$section).Attributes["key"].Value.Equals("Password"))
```

Comment:
Guessing you meant for "Password" to be exact case match. It's not specified, so just wanted to make sure it wasn't oversight.